Source/WebKit/GPUProcess/media/RemoteSourceBufferProxy.cpp

```diff
@@ -129,14 +127,14 @@ void RemoteSourceBufferProxy::sourceBufferPrivateHighestPresentationTimestampCha
     m_connectionToWebProcess->connection().send(Messages::SourceBufferPrivateRemote::SourceBufferPrivateHighestPresentationTimestampChanged(timestamp), m_identifier);
 }
 
-void RemoteSourceBufferProxy::sourceBufferPrivateDurationChanged(const MediaTime& duration, CompletionHandler<void()>&& completionHandler)
+Ref<GenericPromise> RemoteSourceBufferProxy::sourceBufferPrivateDurationChanged(const MediaTime& duration)
 {
-    if (!m_connectionToWebProcess) {
-        completionHandler();
-        return;
-    }
+    if (!m_connectionToWebProcess)
+        return GenericPromise::createAndReject(-1);
 
-    m_connectionToWebProcess->connection().sendWithAsyncReply(Messages::SourceBufferPrivateRemote::SourceBufferPrivateDurationChanged(duration), WTFMove(completionHandler), m_identifier);
+    return m_connectionToWebProcess->connection().sendWithPromisedReply(Messages::SourceBufferPrivateRemote::SourceBufferPrivateDurationChanged(duration), m_identifier)->whenSettled(RunLoop::main(), [] {
```

Comment:
Seems like `sendWithPromisedReply(...)` should have a variant that doesn't take a lambda, and just creates and returns a `GenericPromise` with the result.

Reply:
`sendWithPromisedReply` doesn't take anything, it returns a NativePromise. You don't have to act on the promise

Reply:
The issue is returning the IPC error code; this wasn't handled before, and code always assumed that sending an IPC message was successful. We needed a way to pass that error. Merging the error code with the PlatformMediaError would be nice, but I don't know how that will play for consistency.

`sendWithMediaPromisedReply` could do that maybe